### DISCUSSION TITLE
Visual C++ .vcproj files?

### DISCUSSION THREAD

**Discussion by wsu-cb** [wxFormBuilder/wxFormBuilder#765] [Q&A]
How difficult would it be it be (for me) to create Microsoft Visual C++ .vcproj files by hand rather than using CMake?  I am wondering how practical it would be to make wxFormBuilder be a git submodule to my own project, and build both wxFormBuilder and the main project using a single .sln without needing any dependency on CMake.

**Comment by sodevel** [comment]
How well can you read CMake files and recreate the logic there or read MSVC Project Files and rebuild them without CMake calls? That determines how difficult it would be for you.

In general, wxFB is designed to be used as git submodule and as CMake subproject, mostly to simplify the development of plugins, but this can also be used for other purposes. I put some effort into the CMake files to produce good looking MSVC Project files.

Creating MSVC Project files manually will be a lot of work, keeping them working after future updates too. Beside the compiling of the sources, the next biggest task is the processing of the resources to get a working file layout. A lot of files need to be copied into the proper locations, in the future additional processing might be necessary. Another build task is the extraction of git metadata to include them into source files.

I cannot recommend that approach, if your project is not using CMake as build system, i would investigate how you can integrate the CMake generated MSVC Project files of wxFB instead.

**Reply by wsu-cb** [reply]
Could you please clarify:

Are you saying that the CMake-generated MSVC project files are already committed to the git repository?  That would be great, but I'm not seeing them right now.

Or do you mean, even if my main project doesn't use CMake, if I want to compile wxFormBuilder, I still effectively have a dependency on CMake in order to generate the MSVC project files?

**Reply by sodevel** [reply]
No, MSVC project files generated by CMake cannot be added to the repository, they will contain absolute paths and other information dependent on the machine on which they got generated.

Yes, i recommend to keep the dependency on CMake to generate the project files, this should be more robust than writing project files yourself. I'm not that familiar with MSVC project files, i don't know if u can have multiple .sln files in a project or what such file contains exactly. If it only contains links to .vcxproj files you don't need the generated file and could just link to the generated .vcxproj files from your .sln file.

Also, CMake is not only required to generate the project files, it is also required during the build itself.

**Reply by wsu-cb** [reply]
Thank you for the information.  I was trying to avoid adding additional pre-compiled dependencies to my project, but it looks like there's no way around it.  As such, I may as well go directly to depending on wxFormBuilder's binaries rather than CMake.

**Comment by tester0077** [comment]
For my way of generating MSVC 2022 sln files, please see my post on the wxWidgets forum:
https://forums.wxwidgets.org/viewtopic.php?f=19&t=50818
